### ISSUE TITLE
sam-dump prefix not added to unmapped reads (serious bug)

### ISSUE THREAD

**Issue by SooLee** [ncbi/sra-tools#422]
When `sam-dump` (version 2.10.7) is run with the `-u -p <prefix>` command does not add the prefix to the unmapped reads. It results in a bam file as below, and when fastq is extracted it causes unmatching read problems which is hard to detect because the two fastq files are of the same length and most of the reads are in the same order when you open the files to check, but down the road when you realign these reads using bwa-mem, every single run gives you an error of unmatching reads. Some people could waste thousands of dollars on their batch processes on the cloud because of this bug. This is a pretty serious bug and needs to be fixed asap:

```
SRR10148433.215172828	153	9	35489590	0	7S30M114S	*	0	0	CCGCACCACCCCACCTCACCCCACCCCACTCCACCCCTAAGCTCACACCAACCAGCCTCATCCTTTCAAACATCATTTTTTTTAATGATACGGCGACCACCGAGATCTACACAACTTGATACACTCTTTCCCTACACGACGCTCTTCCGAT	-))7)-))-))--)))))7-)7)7)A77-----7-7----7-7--7-7<--7---7--7-7<A7---<---7--<-<JJJJJJJJJJJJFJJAJJJJJJJJJJJJJFFJJJJJJJJJJJJJJFJFJJJJJJJJJF<JJJJJJJJJJFFAAA	RG:Z:2896051402	NH:i:1	NM:i:0
215172828	101	*	0	0	*	9	35489590	0	ATCGGAAGAGCACACGTCTGAACTCCAGTCACAACTTGATATCTCGTATGCCGTCTTCTGCTTGAAAAAAAAACACAGGTAGAAAACTATACGGCGTAATGGAGGAAGTGAAGAGAACTATCTGATAGGTCCGCCCATAGTTACGTTTGTA	AAFFFJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJFJJJJJJJJJFJJJJJJJJJJFJJJJJ-------<--7-----7-A<-------------7----7--------77----7-7-77---7-7----------------	RG:Z:2896051402
```

**Comment by durbrow** [comment]
You are correct this is a bug. But unless you are mixing together data from multiple accessions, there's no need to prefix the spot number with the accession. It is not difficult to prepend the accession name to the output from sam-dump. For example, using awk:
`sam-dump -u SRR10148433 | awk '!/^@/ { print "SRR10148433."$0 } /^@/ {print}'`
will prepend the accession to every line that is not a header line.
